File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> valToIndex = new HashMap<>();
        
        for(int i = 0; i < nums.length; i++){
            int diff = target - nums[i];
            if(valToIndex.containsKey(diff)){
                return new int[]{valToIndex.get(diff), i};
            }
            
            valToIndex.put(nums[i], i);
        }
        
        return new int[]{};
    }
}

File: 1-two-sum/two-sum.ts
function twoSum(nums: number[], target: number): number[] {
    const diff = {};
    for (let i = 0; i < nums.length; i++) {
        if (diff[target - nums[i]] != null && diff[target - nums[i]] !== i) {
            return [diff[target - nums[i]], i];
        }
        diff[nums[i]] = i;
    }

    return [-1, -1];
};